Fixed a bug on line 211 from setTime to setTitle

RoguedBear opened the pull request:
if a difference in titles was detected between GCal and Toggl entries, the lines 210-211 were executed. 
Fixed the function from `setTime()` to `setTitle()`, which otherwise gave an error for me at least when passing 'description' to `setTime()`

RoguedBear committed:
Fixed a bug on lin 211 from setTime to setTitle

if a difference in titles was detected between GCal and Toggl entries, the lines 210-211 were executed. 
Fixed the function from `setTime()` to `setTitle()`, which otherwise gave an error for me at least when passing 'description' to `setTime()`

ProExpertProg reviewed (approved):
Yeah the script has been failing for me but I never took the time to look why. Thanks for catching this.

On another note, I'm glad this can be useful to someone other than me :D

RoguedBear commented:
> Yeah the script has been failing for me but I never took the time to look why. Thanks for catching this.
> 
> On another note, I'm glad this can be useful to someone other than me :D

Thank you for making this script! It really has been useful for me, now that I have started to track my time again and now that Zapier has also added 100 tasks limit, this script is all I needed.

I have also made some other modifications to the script:

- When getting all events from Toggl, instead of fetching it from last year, I've made it to fetch it from 1 day ago.
  1 day seems a reasonable time range to me instead of 1 year. less execution time, less headache 😅
- Similarly for fetching GCal events, I've fetched events from 2 days ago instead of 1. 
  - When fetching events from 1 day, the thing that happened was, for the same time range Toggl was sending all events from 12am of the previous day to the current time (say for example, 27 entries). 

  - But, GCal was sending events within 24hr time range instead of events from 12am of the starting of the  previous day (GCal sent, 21 events for the same time range)
  So to fix that, GCal now fetches events from 2 days prior to the current day, so now we have more GCal events than Toggl entries.
This prevents duplicating of events which was happening with me when I made this change later on in the day.
Your Logging code (Toggl entries length, GCal events length) helped me figure out why script was duplicating events in Calendar

- modified the if/else in else part (line [201](https://github.com/ProExpertProg/Toggl-and-Google-calendar-integration/blob/ccca7132c64536b553451f180d78c888c97a5e52/Code.gs#L201)) condition where if I inadvertently change any of the properties (title, time, description) of the events in my GCal, the program will run the modification part and fix it back.
- and because of my previous change I added more detail to `Logging` in that part to show which property of the event was fixed.
- Purely subjective: in the GCal event title, after the name of the event I've added code to also write the duration too
  example: Before calendar event title: `Used Computer`
 After calendar event title: `Used Computer in 4hr 13min`

(Thanks for adding the comments and the stats that you were logging, they were really helpful!)
